xarray/backends/rasterio_.py

```diff
@@ -91,11 +93,32 @@ def open_rasterio(filename):
     -------
     data : DataArray
         The newly created DataArray.
+    chunks : int, tuple or dict, optional
+        Chunk sizes along each dimension, e.g., ``5``, ``(5, 5)`` or
+        ``{'x': 5, 'y': 5}``. If chunks is provided, it used to load the new
+        DataArray into a dask array. This is an experimental feature; see the
+        documentation for more details.
+    cache : bool, optional
+        If True, cache data loaded from the underlying datastore in memory as
+        NumPy arrays when accessed to avoid reading from the underlying data-
+        store multiple times. Defaults to True unless you specify the `chunks`
+        argument to use dask, in which case it defaults to False. Does not
+        change the behavior of coordinates corresponding to dimensions, which
+        always load their data from disk into a ``pandas.Index``.
+    lock : False, True or threading.Lock, optional
+        If chunks is provided, this argument is passed on to
+        :py:func:`dask.array.from_array`. By default, a per-variable lock is
+        used when reading data from netCDF files with the netcdf4 and h5netcdf
```

Comment:
OK, I am on it.

@shoyer , I have trouble to understand what's going on in the ``open_dataset`` workflow. The documentation says: "By default, a per-variable lock is used when reading data from netCDF files with the netcdf4 and h5netcdf engines", but then the lock is decided once [here](https://github.com/pydata/xarray/blob/master/xarray/backends/api.py#L310). This will default to ``GLOBAL_LOCK`` and then passed to ``maybe_decode_store``. Where is the "per variable" logic happening? Or is the ``GLOBAL_LOCK`` already doing this? (it is defined [here](https://github.com/pydata/xarray/blob/371d034372bc7522098a142a0debf93916c49102/xarray/backends/common.py#L30)) 

(excuse my poor knowledge about these issues)